csrc/backend_ops/tensorrt/CMakeLists.txt

```diff
@@ -3,41 +3,42 @@ cmake_minimum_required(VERSION 3.14)
 include(${CMAKE_SOURCE_DIR}/cmake/cuda.cmake NO_POLICY_SCOPE)
 project(mmdeploy_tensorrt_ops CUDA CXX)
 
-include(${CMAKE_SOURCE_DIR}/cmake/common.cmake)
+include(${CMAKE_SOURCE_DIR}/cmake/MMDeploy.cmake)
 include(${CMAKE_SOURCE_DIR}/cmake/cuda.cmake NO_POLICY_SCOPE)
 include(${CMAKE_SOURCE_DIR}/cmake/tensorrt.cmake NO_POLICY_SCOPE)
 
-set_targets(${PROJECT_NAME} BACKEND_OPS_OBJ BACKEND_OPS_STATIC BACKEND_OPS_MODULE)
-
 # cub
 if (NOT DEFINED CUB_ROOT_DIR)
     if (CUDA_VERSION VERSION_LESS 11.0)
         set(CUB_ROOT_DIR "${CMAKE_SOURCE_DIR}/third_party/cub")
-    endif()
+    endif ()
 endif ()
 
 file(GLOB_RECURSE BACKEND_OPS_SRCS *.cpp *.cu)
-build_object_target(${BACKEND_OPS_OBJ} "${BACKEND_OPS_SRCS}")
-target_compile_definitions(${BACKEND_OPS_OBJ}
+add_library(${PROJECT_NAME}_obj OBJECT "${BACKEND_OPS_SRCS}")
+set_target_properties(${PROJECT_NAME}_obj PROPERTIES POSITION_INDEPENDENT_CODE 1)
+target_compile_definitions(${PROJECT_NAME}_obj
         PRIVATE -DTHRUST_IGNORE_DEPRECATED_CPP_DIALECT=1)
-target_include_directories(${BACKEND_OPS_OBJ}
+target_include_directories(${PROJECT_NAME}_obj
         PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/common)
-target_include_directories(${BACKEND_OPS_OBJ}
+target_include_directories(${PROJECT_NAME}_obj
         PRIVATE ${CUDA_TOOLKIT_ROOT_DIR}/include)
-target_include_directories(${BACKEND_OPS_OBJ} PRIVATE ${TENSORRT_INCLUDE_DIR})
-target_include_directories(${BACKEND_OPS_OBJ} PRIVATE ${CUDNN_DIR}/include)
-target_include_directories(${BACKEND_OPS_OBJ} PRIVATE ${CUB_ROOT_DIR})
-target_link_directories(${BACKEND_OPS_OBJ} PUBLIC ${CUDNN_DIR}/lib64)
-target_link_libraries(${BACKEND_OPS_OBJ}
+target_include_directories(${PROJECT_NAME}_obj PRIVATE ${TENSORRT_INCLUDE_DIR})
+target_include_directories(${PROJECT_NAME}_obj PRIVATE ${CUDNN_DIR}/include)
+target_include_directories(${PROJECT_NAME}_obj PRIVATE ${CUB_ROOT_DIR})
+target_link_directories(${PROJECT_NAME}_obj PUBLIC ${CUDNN_DIR}/lib64 ${CUDNN_DIR}/lib/x64)
+target_link_libraries(${PROJECT_NAME}_obj
         PRIVATE ${TENSORRT_LIBRARY} ${CUDA_CUBLAS_LIBRARIES} cudnn)
+mmdeploy_export(${PROJECT_NAME}_obj)
 
 # Build module library. It is used to convert onnx model to tensorrt engine
-build_module_target(${BACKEND_OPS_MODULE} ${BACKEND_OPS_OBJ} "PRIVATE")
-add_library(mmdeploy::tensorrt_ops ALIAS ${BACKEND_OPS_MODULE})
-install_targets(${BACKEND_OPS_MODULE})
+mmdeploy_add_module(${PROJECT_NAME} SHARED $<TARGET_OBJECTS:${PROJECT_NAME}_obj>)
```

Comment:
Neither cublas nor cudnn is found  when executing `ldd libmmdeploy_tensorrt_ops.so`